tests/positive/files/units.go

```diff
@@ -114,3 +116,147 @@ func CreateEmptyDropinsUnit() types.Test {
 		ConfigMinVersion: configMinVersion,
 	}
 }
+
+// TestUnmaskUnit tests unmasking systemd units. In the case that a
+// systemd unit is masked already it will unmask it. In the case it is
+// not masked it will do nothing. In the code below we have foo.service,
+// which is masked in the input, and bar.service, which is not masked in
+// the input. We test that foo.service is properly unmasked and
+// enabled and that the unmasking of bar.service is skipped (because
+// it wasn't masked to begin with) and that bar.service gets enabled.
+func TestUnmaskUnit() types.Test {
+	name := "unit.unmask"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": { "version": "$version" },
+		"systemd": {
+			"units": [
+				{
+					"name": "foo.service",
+					"enabled": true,
+					"mask": false
+				},
+				{
+					"name": "bar.service",
+					"enabled": true,
+					"mask": false
+				}
+			]
+        }
+		}`
+	configMinVersion := "3.0.0"
+	in[0].Partitions.AddLinks("ROOT", []types.Link{
+		{
+			Node: types.Node{
+				Directory: "etc/systemd/system",
+				Name:      "foo.service",
+			},
+			Target: "/dev/null",
+			Hard:   false,
+		},
+	})
+	in[0].Partitions.AddFiles("ROOT", []types.File{
+		{
+			Node: types.Node{
+				Name:      "foo.service",
+				Directory: "usr/lib/systemd/system",
+			},
+			Contents: "[Unit]\nDescription=f\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n[Install]\nWantedBy=multi-user.target\n",
+		},
+		{
+			Node: types.Node{
+				Name:      "bar.service",
+				Directory: "usr/lib/systemd/system",
+			},
+			Contents: "[Unit]\nDescription=f\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n[Install]\nWantedBy=multi-user.target\n",
+		},
+	})
+	out[0].Partitions.AddFiles("ROOT", []types.File{
+		{
+			Node: types.Node{
+				Name:      "20-ignition.preset",
+				Directory: "etc/systemd/system-preset",
+			},
+			Contents: "enable foo.service\nenable bar.service\n",
```

Comment:
unfortunately the order of these entries this seems to be non-deterministic. on subsequent runs I can get `PASS` and then:

```
validator.go:234: Contents of file /var/tmp/ignition-blackbox-997495071/hd0p9/etc/systemd/system-preset/20-ignition.preset do not match! enable foo.service
    enable bar.service
     enable bar.service
    enable foo.service

```

any ideas?

Reply:
No clever ones.  You could drop `enabled: true` from some of the cases, or split up tests that enable multiple units.

Reply:
ok I dropped the `enabled: true` from one of the units.